### DISCUSSION TITLE
Suggestion: Maintain the original live server, they are looking for a maintainer.

### DISCUSSION THREAD

**Discussion by douira** [yandeu/five-server#25] [General]
The [original live server project](https://github.com/ritwickdey/vscode-live-server) has a notice on their VSCode Marketplace page stating that they're looking for a maintainer. I suggest the maintainers of this fork apply as maintainers there so that that project can continue to be helpful since it has 14 million downloads and is well-known. (instead of creating a niche fork that might fix bugs, but it barely does any good because nobody knows it) I'm not trying to shut you down or anything, just suggesting that if there's people here who are interested in improving and maintaining live server, that work is much more valuable in the original project where it's needed.

**Comment by yandeu** [comment]
If @ritwickdey decides to not maintain Live Server anymore, I would prefer he adds a depreciation message and links to Five Server.

But I don't know what @ritwickdey's plans are :/

_Btw, same applies for @tapio._

**Reply by douira** [reply]
There is a message on the [vscode marketplace page](https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer) that states

> [I'm sorry but I'm super busy now. If you want to be a maintainer of the project, please feel free to contact me! You've to be passionate about programming]

I'd prefer if the original project with 14.2 million downloads was maintained by you and the other people who run this fork instead of pushing for the adoption of a fork with a slightly different name. The maintainer of live-server is obviously interested in handing the project over to new maintainers so there's really no need for a fork IMO. You probably just need to ask him and you could pull your changes from this project into live-server. I think handing over maintenance is better than creating maintained forks because it creates less confusion and duplication of projects.

**Reply by yandeu** [reply]
Well, `VSCode Five Server` is NOT a fork of `VSCode Live Server`.

`VSCode Live Server` is based on `Live Server (npm)`.

`VSCode Five Server` is based on `Five Server (npm)`, which is not a fork of `Live Server (npm)`, but a complete re-write.

`VSCode Five Server` does not use a single line from `VSCode Live Server`.

So the word "fork", is a bit missleading.

**Reply by douira** [reply]
The fact that this project describes itself as `Maintained Fork of Live Server` in the second line of its readme makes this very confusing. Usually fork means that the original project's code was used.

**Reply by yandeu** [reply]
That is the case. `Live Server (npm)` was taken as a basis.
Many files still hold the license of `Live Server (npm)`.

**Reply by douira** [reply]
So you expanded on the original code or was no original code used? Given that you say there is a license from the previous project then I wouldn't call it a "rewrite" but an actual fork. This is getting into the semantics of the words though. Either way, my point still stands. I'm in favour of continued development on the original project (even by non-original contributors) instead of mostly unknown derivatives (rewrites, forks, whatever). But this is your project so you do you.

**Reply by yandeu** [reply]
For example, compare these 2 files. They still have a similar structure:
- https://github.com/yandeu/five-server/blob/main/src/index.ts
- https://github.com/tapio/live-server/blob/master/index.js

**Comment by Danny-Engelman** [comment]
@yandue, at the start of your fork you mentioned a one-week update cycle.
Last update was 4 months ago, do you intend to maintain and update this project?

**Reply by yandeu** [reply]
Yes. The last commit was 26 days ago.  
And the last update to the VSCode Marketplace was 51 days ago.

**Comment by trusktr** [comment]
This fork cannot be used for anything based on the license, and is closed source:

- https://github.com/yandeu/five-server/discussions/116
